src/storage/v3/shard.cpp

```diff
@@ -1022,15 +1017,13 @@ void Shard::CollectGarbage(const io::Time current_time) {
     RemoveObsoleteEntries(&indices_, clean_up_before_timestamp);
   }
 
-  auto vertex_acc = vertices_.access();
   for (const auto &vertex : deleted_vertices_) {
-    MG_ASSERT(vertex_acc.remove(vertex), "Invalid database state!");
+    MG_ASSERT(vertices_.erase(*vertex), "Invalid database state!");
   }
```

Comment:
```suggestion
  for (const auto *vertex : deleted_vertices_) {
    MG_ASSERT(vertices_.erase(*vertex), "Invalid database state!");
  }
```

Do not take a reference to a pointer, but copy it.